From 1b32a0e398871f0f5b7b41d200b1b9db371e84d5 Mon Sep 17 00:00:00 2001
From: Nikolai Kondrashov <Nikolai.Kondrashov@redhat.com>
Date: Mon, 15 Dec 2014 16:48:26 +0200
Subject: [PATCH 1/1] raddb: Move trigger.conf INCLUDE before modules

Move "$INCLUDE trigger.conf" chunk before module section in
"radiusd.conf.in". This makes it possible to reference "snmptrap" and
related trigger variables under "pool.trigger" in module configurations,
simplifying them.

E.g. like this (in raddb/mods-enabled/ldap):

ldap ldap_instance {
    pool {
        trigger {
            args = "radiusdModuleName s '${...:name}' radiusdModuleInstance s '${...:instance}'"
            open = "${snmptrap}::serverModuleConnectionUp ${args}"
            close = "${snmptrap}::serverModuleConnectionDown ${args}"
        }
    }
}
---
 raddb/radiusd.conf.in | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/raddb/radiusd.conf.in b/raddb/radiusd.conf.in
index c6d5834..fb7d1bd 100644
--- a/raddb/radiusd.conf.in
+++ b/raddb/radiusd.conf.in
@@ -648,6 +648,14 @@ thread pool {
 	auto_limit_acct = no
 }
 
+######################################################################
+#
+#  SNMP notifications.  Uncomment the following line to enable
+#  snmptraps.  Note that you MUST also configure the full path
+#  to the "snmptrap" command in the "trigger.conf" file.
+#
+#$INCLUDE trigger.conf
+
 # MODULE CONFIGURATION
 #
 #  The names and configuration of each module is located in this section.
@@ -780,14 +788,6 @@ policy {
 
 ######################################################################
 #
-#  SNMP notifications.  Uncomment the following line to enable
-#  snmptraps.  Note that you MUST also configure the full path
-#  to the "snmptrap" command in the "trigger.conf" file.
-#
-#$INCLUDE trigger.conf
-
-######################################################################
-#
 #	Load virtual servers.
 #
 #	This next $INCLUDE line loads files in the directory that
-- 
2.1.3

